pkg/auth/client/impersonate.go

```diff
@@ -72,3 +62,51 @@ func NewImpersonatingKubernetesClientset(user user.Info, config restclient.Confi
 	impersonatingConfig := NewImpersonatingConfig(user, config)
 	return kclientset.NewForConfig(&impersonatingConfig)
 }
+
+// impersonatingRESTClient sets impersonating user, groups, and scopes headers per request
+type impersonatingRESTClient struct {
+	user     user.Info
+	delegate restclient.Interface
+}
+
+func NewImpersonatingRESTClient(user user.Info, client restclient.Interface) restclient.Interface {
+	return &impersonatingRESTClient{user: user, delegate: client}
+}
+
+// Verb does the impersonation per request by setting the proper headers
+func (c impersonatingRESTClient) Verb(verb string) *restclient.Request {
```

Comment:
It's been a while, but if I recall correctly it required access to internal Request structure members or methods not exposed by the module or something simialr. It was a while ago so I do not recall the exact issue, but it was too invasive to go that way.